tests/encryption/test_intransit_encryption_sanity.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.resources.storage_cluster import (
+    in_transit_encryption_verification,
+    set_in_transit_encryption,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_intransit_encryption_notset,
+    tier1,
+    skipif_ocs_version,
+)
+
+log = logging.getLogger(__name__)
+
+
+@skipif_intransit_encryption_notset
+class TestInTransitEncryptionSanity:
+    def teardown(self):
+        set_in_transit_encryption()
+
+    @tier1
+    @skipif_ocs_version("<4.12")
+    @pytest.mark.polarion_id("OCS-4861")
+    def test_intransit_encryption_enable_disable_statetransition(self, request):
+        """
+        The test does the following:
+        1. Verify in-transit Encryption is Enable on setup.
+        2. Disable Encryption
+        3. Verify in-transit encryption configuration is removed.
+        4. Enable encryption Again and verify it.
+        5. Verify in-transit encryption config is exists.
+
+        """
+        log.info("Verifying the in-transit encryption is enable on setup.")
+        assert in_transit_encryption_verification()
```

Comment:
why not to place the `addfinalizer` in the `teardown` fixture itself?
For example https://github.com/red-hat-storage/ocs-ci/blob/6a2f2fa9c3c37bfd6f36dffd16896c35b19cdce6/tests/e2e/kcs/test_noobaa_db_backup_and_recovery.py#L86

Reply:
Moved addfinalizer in trardown fixture.